lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2231,6 +2231,22 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
     // For llvm.fmuladd.* fusion is not guaranteed. If a fused multiply-add
     // is required the corresponding llvm.fma.* intrinsic function should be
     // used instead.
+    if (!checkTypeForSPIRVExtendedInstLowering(II, BM))
```

Comment:
There is no handling for `fmuladd` in `checkTypeForSPIRVExtendedInstLowering`, it will always return `true`. Should we extend `checkTypeForSPIRVExtendedInstLowering`?

Reply:
I don't think that this is necessary, because `llvm.fmuladd.*` is only defined for floating-point scalars, which perfectly matches requirements of `fma` extinst. We don't even need to check for vector length, because `llvm.fmuladd.*` is not defined for vectors

Reply:
`checkTypeForSPIRVExtendedInstLowering` takes `II` as a parameter. This function is based on `switch` on intrinsic id and there is no `case` for `Intrinsic::fmuladd`. We either should add `case Intrinsic::fmuladd:`, or don't call `checkTypeForSPIRVExtendedInstLowering` at all.

Reply:
Fair enough, I will remove the call